src/index.js

```diff
@@ -1,7 +1,13 @@
 window.onload = function() {
 	var ajax = new XMLHttpRequest();
 	ajax.open("GET", "banner.html");
-	ajax.responseType = "document";
 	ajax.send();
-	document.querySelector("#banner").innerHTML += ajax.responseXML;
-}
\ No newline at end of file
+	// When the ajax request changes state
```

Comment:
This was the root of the problem. Basically, you needed to wait for the request to be complete before attempting to inject the response. By waiting for the state to reach state 4, you know that you have a complete response and then you can do the processing on it.